lib/checks/hasUnsupportedNodes.js

```diff
@@ -1,30 +1,29 @@
-const Node = require('../node');
+const {query} = require('math-nodes');
+
 const resolvesToConstant = require('./resolvesToConstant');
 
 function hasUnsupportedNodes(node) {
-  if (Node.Type.isParenthesis(node)) {
```

Comment:
Oh - good question. It's used to return early and just not do anything.

We send any user input from Socratic to mathjs to try to parse it. If it fails to parse, we don't do anything. If it parses but has mathjs nodes we don't support we also don't do anything. And then we assume that all the mathjs trees we're working with only have one of function/parenthesis/constant/etc nodes (the ones you see around the codebase)

Let me know if I can make that more clear